lib/secretsscanner/authorizedkeys/authorized_keys.go

```diff
@@ -105,19 +106,16 @@ func NewWatcher(ctx context.Context, config WatcherConfig) (*Watcher, error) {
 	if config.Clock == nil {
 		config.Clock = clockwork.NewRealClock()
 	}
-	if config.etcPasswdFile == "" {
-		// etcPasswordPath is the path to the password file.
-		// This file is used to get the list of users on the system and their home directories.
-		const etcPasswordPath = "/etc/passwd"
-		config.etcPasswdFile = etcPasswordPath
+	if config.getHostUsers == nil {
+		config.getHostUsers = getHostUsers
 	}
 
 	w := &Watcher{
-		client:           config.Client,
-		logger:           config.Logger,
-		clock:            config.Clock,
-		hostID:           config.HostID,
-		usersAccountFile: config.etcPasswdFile,
+		client:       config.Client,
+		logger:       config.Logger,
+		clock:        config.Clock,
+		hostID:       config.HostID,
+		getHostUsers: config.getHostUsers,
```

Comment:
```suggestion
		getHostUsers: cmp.Or(config.getHostUsers, getHostUsers),
```

Then you can eliminate the nil check on line 109.

Reply:
Functions headers aren't comparable